### PR TITLE
Move revenue logic into trackRevenue method, only call if revenue/total exists

### PR DESCRIPTION
**Pull request by ladanazita**

### CODE REVIEW

File: Pod/Classes/SEGAmplitudeIntegration.m [hunk 1]
```diff
@@ -77,56 +77,61 @@ - (void)realTrack:(NSString *)event properties:(NSDictionary *)properties integr
...
+        }
+        id quantity = [properties objectForKey:@"quantity"];
+        if (!quantity || ![quantity isKindOfClass:[NSNumber class]]) {
+            quantity = [NSNumber numberWithInt:1];
+        }
+        id receipt = [properties objectForKey:@"receipt"];
+        if (!receipt || ![receipt isKindOfClass:[NSString class]]) {
+            receipt = nil;
         }
+        NSLog(@"Number : %@", revenue);
```
Comment (f2prateek) [inline Pod/Classes/SEGAmplitudeIntegration.m]: This is form before, but what does this doo?

File: Pod/Classes/SEGAmplitudeIntegration.m [hunk 2]
```diff
@@ -77,56 +77,61 @@ - (void)realTrack:(NSString *)event properties:(NSDictionary *)properties integr
...
-                                 price:revenue
-                               receipt:receipt];
+        [self trackRevenue:properties andRevenue:revenue];
+    }
+}
+
+- (void)trackRevenue:(NSDictionary *)properties andRevenue:(NSNumber *)revenue
+{
+    // Use logRevenueV2 with revenue properties.
+    if ([(NSNumber *)[self.settings objectForKey:@"useLogRevenueV2"] boolValue]) {
```
Comment (f2prateek) [inline Pod/Classes/SEGAmplitudeIntegration.m]: Use an early return here?

```
if (![(NSNumber *)[self.settings objectForKey:@"useLogRevenueV2"] boolValue]) {
 ...
}

// previously nested code goes ehre.
```

Comment (ladanazita) [reply Pod/Classes/SEGAmplitudeIntegration.m]: I was going to refactor the logic in `trackRevenue` in a separate PR

File: Pod/Classes/SEGAmplitudeIntegration.m (hunk 1 above)
Comment (ladanazita) [reply Pod/Classes/SEGAmplitudeIntegration.m]: I removed it in https://github.com/segment-integrations/analytics-ios-integration-amplitude/pull/35

**Review by f2prateek — APPROVED** [review]

**Pull request merged**